src/services/aave.js

```diff
@@ -0,0 +1,244 @@
+    const lendingPoolContract = await this.getLendingPoolContract();
+    if (!lendingPoolContract) return Promise.resolve({});
+
+    const depositedAssetData = await lendingPoolContract
+      .getUserReserveData(asset.address, accountAddress)
+      .catch(() => ([]));
+
+    const earnInterestRateBN = depositedAssetData[5];
+    const earnInterestRate = rayToNumeric(earnInterestRateBN) * 100; // %
+    const currentBalanceBN = depositedAssetData[0];
+    const currentBalance = Number(formatAmount(utils.formatUnits(currentBalanceBN, asset.decimals)));
+
+    let earnedAmount = 0;
+    let initialBalance = 0;
+    const aaveTokenContract = await this.getAaveTokenContractForAsset(asset.address);
+    if (aaveTokenContract) {
+      const initialBalanceBN = await aaveTokenContract.principalBalanceOf(accountAddress);
+      const earnedAmountBN = currentBalanceBN.sub(initialBalanceBN);
+      initialBalance = Number(formatAmount(utils.formatUnits(initialBalanceBN, asset.decimals)));
+      earnedAmount = Number(formatAmount(utils.formatUnits(earnedAmountBN, asset.decimals)));
+    }
+
+    // percentage gain formula
+    const earningsPercentageGain = ((currentBalance - initialBalance) / initialBalance) * 100;
+
+    return Promise.resolve({
+      ...asset,
+      earnInterestRate,
+      currentBalance,
+      earnedAmount,
+      earningsPercentageGain,
+      initialBalance,
+      aaveTokenAddress: aaveTokenContract?.address,
+    });
+  }
+
+  async getAccountDepositedAssets(
+    accountAddress: string,
+    accountAssets: Asset[],
+    supportedAssets: Asset[],
+  ): Promise<DepositedAsset[]> {
+    const supportedDeposits = await this.getSupportedDeposits(accountAssets, supportedAssets);
+    const depositedAssets = await Promise.all(supportedDeposits.map((asset) => {
+      return this.fetchAccountDepositedAsset(accountAddress, asset);
+    }));
+    return Promise.resolve(depositedAssets.filter(({ initialBalance }) => !!initialBalance));
+  }
+
+  async fetchAccountDepositAndWithdrawTransactions(accountAddress: string): Promise<Object> {
+    const url = `https://api.thegraph.com/subgraphs/id/${AAVE_THE_GRAPH_ID}`;
+    return axios
+      .post(url, {
+        timeout: 5000,
+        query: `
+        {
+          deposits (orderBy: timestamp, orderDirection: desc, where: { 
+            user: "${accountAddress.toLowerCase()}"
+          }) {
+            id
+            amount
+            reserve {
+              symbol
+              decimals
+            }
+          }
+          withdraws: redeemUnderlyings (orderBy: timestamp, orderDirection: desc, where:{
+            user: "${accountAddress.toLowerCase()}"
+          }) {
+            id
+            amount
+            reserve {
+              symbol
+              decimals
+            }
+          }
+        }
+      `,
+      })
+      .then(({ data: response }) => response.data)
+      .catch(() => ({}));
```

Comment:
Same as above - i think it's definitely needed here:
`Sentry.captureException(err);`